src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1260,6 +1260,65 @@ internal virtual FieldSymbol FixedElementField
         /// <returns>True if this is an interface type.</returns>
         internal abstract bool IsInterface { get; }
 
+        /// <summary>
+        /// Verify if the given type can be used to back a tuple type and return cardinality of that tuple type in <paramref name="tupleCardinality"/>. 
+        /// </summary>
+        /// <param name="tupleCardinality">If method returns true, contains cardinality of the compatible tuple type.</param>
+        /// <returns></returns>
+        public override bool IsTupleCompatible(out int tupleCardinality)
+        {
+            if (IsTupleType)
+            {
+                tupleCardinality = TupleElementTypes.Length;
+                return true;
+            }
+
+            // Should this be optimized for perf (caching for VT<0> to VT<7>, etc.)?
+            if (!IsUnboundGenericType &&
+                ContainingSymbol.Kind == SymbolKind.Namespace &&
+                Name == TupleTypeSymbol.TupleTypeName && ContainingNamespace.Name == "System" &&
```

Comment:
I'll see what would be the appropriate place for such a constant.

---
In reply to: [61795672](https://github.com/dotnet/roslyn/pull/11006#discussion_r61795672) [](ancestors = 61795672)